### PR TITLE
Fix crashing when typing in console

### PR DESCRIPTION
**Pull request by Brentdevent**
File contains now a carriage return after downloading, it caused some issues.

### CODE REVIEW

**Review by momo5502 — APPROVED** [review]

**Review by diamante0018 — COMMENTED** [review]
std::ranges::remove, std::ranges::remove_if?

**Review by momo5502 — APPROVED** [review]

**Review by momo5502 — CHANGES_REQUESTED** [review]
https://stackoverflow.com/a/64330309/8725381

File: src/client/component/dvars.cpp [hunk 1]
```diff
@@ -65,7 +65,7 @@ namespace dvars
 				return;
 			}
 
-			data.erase(std::remove(data.begin(), data.end(), '\r'), data.end());
+			data.erase(std::ranges::remove(data, '\r').begin(), data.end());
```
Comment (diamante0018) [inline src/client/component/dvars.cpp:68]: If my understanding is correct this isn't right anymore.
It should be:
`const auto ret = std::ranges::remove(data, ' \r')`
Followed by
`data.erase(ret.begin(), ret.end());`
Please verify and apply changes if needed

Comment (diamante0018) [reply src/client/component/dvars.cpp:68]: Of course the variable names are at your discretion, I just typed an example of what I interpreted to he the correct usage from my phone :P

Comment (diamante0018) [reply src/client/component/dvars.cpp:68]: Alternatively also the compound statement (I believe that's what it is called), is available:
```
const auto [beg, end] = std::ranges::remove(data, ' \r');
data.erase(beg, end);
```

Comment (Brentdevent) [reply src/client/component/dvars.cpp:68]: Got it, I'll go with the last one seems the best option for readability. Thank you!

Comment (momo5502) [reply src/client/component/dvars.cpp:68]: decomposition seems best to me

**Review by momo5502 — APPROVED** [review]

**Review by diamante0018 — APPROVED** [review]
Oops

**Pull request merged**